dev/Pager/PrototypePager/PrototypePager.Events.cs

```diff
@@ -0,0 +1,196 @@
+                case ButtonVisibilityMode.AlwaysVisible:
+                    VisualStateManager.GoToState(this, PreviousPageButtonStates[0], false);
+                    break;
+                case ButtonVisibilityMode.None:
+                    VisualStateManager.GoToState(this, PreviousPageButtonStates[1], false);
+                    break;
+                case ButtonVisibilityMode.HiddenOnEdge:
+                    UpdateHiddenOnEdgeButtons();
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        private void OnFirstPageButtonVisibilityChanged()
+        {
+            switch (this.FirstPageButtonVisibility)
+            {
+                case ButtonVisibilityMode.Auto:
+                case ButtonVisibilityMode.AlwaysVisible:
+                    VisualStateManager.GoToState(this, FirstPageButtonStates[0], false);
+                    break;
+                case ButtonVisibilityMode.None:
+                    VisualStateManager.GoToState(this, FirstPageButtonStates[1], false);
+                    break;
+                case ButtonVisibilityMode.HiddenOnEdge:
+                    UpdateHiddenOnEdgeButtons();
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        private void DisablePageButtonsOnEdge()
+        {
+            if (SelectedIndex == 1)
+            {
+                VisualStateManager.GoToState(this, FirstPageButtonStates[3], false);
+                VisualStateManager.GoToState(this, PreviousPageButtonStates[3], false);
+                VisualStateManager.GoToState(this, NextPageButtonStates[2], false);
+                VisualStateManager.GoToState(this, LastPageButtonStates[2], false);
+            } 
+            else if (SelectedIndex == NumberOfPages)
+            {
+                VisualStateManager.GoToState(this, FirstPageButtonStates[2], false);
+                VisualStateManager.GoToState(this, PreviousPageButtonStates[2], false);
+                VisualStateManager.GoToState(this, NextPageButtonStates[3], false);
+                VisualStateManager.GoToState(this, LastPageButtonStates[3], false);
+            }
+            else
+            {
+                VisualStateManager.GoToState(this, FirstPageButtonStates[2], false);
+                VisualStateManager.GoToState(this, PreviousPageButtonStates[2], false);
+                VisualStateManager.GoToState(this, NextPageButtonStates[2], false);
+                VisualStateManager.GoToState(this, LastPageButtonStates[2], false);
+            }
+            UpdateHiddenOnEdgeButtons();
+        }
+
+        private void UpdateHiddenOnEdgeButtons()
+        {
+            if (this.FirstPageButtonVisibility == ButtonVisibilityMode.HiddenOnEdge)
+            {
+                if (this.SelectedIndex != 1)
+                {
+                    VisualStateManager.GoToState(this, FirstPageButtonStates[0], false);
```

Comment:
>FirstPageButtonStates [](start = 55, length = 21)

I think the arrays of strings are interesting but it makes this line harder to ready. I think I would prefer that the constant have a name which tells me which state it is.

Reply:
I've separated split up the arrays into individual variables!